### PR TITLE
catalog: avoid magnitudes getting printed with all decimals

### PR DESCRIPTION
**Pull request by megies**
### What does this PR do?

Rounds printed magnitude values to one decimal in the Catalog/Event string representation. So far magnitudes get printed with a silly amount of decimals, which looks ugly and doesn't make a lot of sense.

### Why was it initiated?  Any relevant Issues?

*Please fill in*

### PR Checklist
- [x] Correct base branch selected? `master` for new features, `maintenance_...` for bug fixes
- [x] This PR is not directly related to an existing issue (which has no PR yet).
- [x] If the PR is making changes to documentation, docs pages can be built automatically.
      Just remove the space in the following string after the + sign: "+ DOCS"
- [x] If any network modules should be tested for the PR, add them as a comma separated list
      (e.g. `clients.fdsn,clients.arclink`) after the colon in the following magic string: "+TESTS:"
      (you can also add "ALL" to just simply run all tests across all modules)
- [ ] All tests still pass.
- [ ] Any new features or fixed regressions are be covered via new tests.
- [ ] Any new or changed features have are fully documented.
- [x] Significant changes have been added to `CHANGELOG.txt` .
- [ ] First time contributors have added your name to `CONTRIBUTORS.txt` .

### CODE REVIEW

**Comment by megies** [comment]
Example code:

```python
from obspy import read_events

cat = read_events()
cat[0].magnitudes[0].mag = 1.298765432109876543
cat[1].magnitudes = []
cat[2].magnitudes[0].mag = 1.0
print(cat)
print(cat[0])
```

Before:
```
3 Event(s) in Catalog:
2012-04-04T14:21:42.300000Z | +41.818,  +79.689 | 1.2987654321098765 mb | manual
2012-04-04T14:18:37.000000Z | +39.342,  +41.044 | manual
2012-04-04T14:08:46.000000Z | +38.017,  +37.736 | 1.0 ML | manual
Event:	2012-04-04T14:21:42.300000Z | +41.818,  +79.689 | 1.2987654321098765 mb | manual

	            resource_id: ResourceIdentifier(id="quakeml:eu.emsc/event/20120404_0000041")
	             event_type: 'not reported'
	          creation_info: CreationInfo(agency_uri=ResourceIdentifier(id="smi:smi-registry/organization/EMSC"), author_uri=ResourceIdentifier(id="smi:smi-registry/organization/EMSC"), creation_time=UTCDateTime(2012, 4, 4, 16, 40, 50), version='1.0.1')
	    preferred_origin_id: ResourceIdentifier(id="quakeml:eu.emsc/origin/rts/261020/782484")
	 preferred_magnitude_id: ResourceIdentifier(id="quakeml:eu.emsc/NetworkMagnitude/rts/261020/782484/796646")
	                   ---------
	     event_descriptions: 1 Elements
	                origins: 1 Elements
	             magnitudes: 1 Elements
```

After:
```
3 Event(s) in Catalog:
2012-04-04T14:21:42.300000Z | +41.818,  +79.689 | 1.3 mb | manual
2012-04-04T14:18:37.000000Z | +39.342,  +41.044 | manual
2012-04-04T14:08:46.000000Z | +38.017,  +37.736 | 1.0 ML | manual

Event:	2012-04-04T14:21:42.300000Z | +41.818,  +79.689 | 1.3 mb | manual

	            resource_id: ResourceIdentifier(id="quakeml:eu.emsc/event/20120404_0000041")
	             event_type: 'not reported'
	          creation_info: CreationInfo(agency_uri=ResourceIdentifier(id="smi:smi-registry/organization/EMSC"), author_uri=ResourceIdentifier(id="smi:smi-registry/organization/EMSC"), creation_time=UTCDateTime(2012, 4, 4, 16, 40, 50), version='1.0.1')
	    preferred_origin_id: ResourceIdentifier(id="quakeml:eu.emsc/origin/rts/261020/782484")
	 preferred_magnitude_id: ResourceIdentifier(id="quakeml:eu.emsc/NetworkMagnitude/rts/261020/782484/796646")
	                   ---------
	     event_descriptions: 1 Elements
	                origins: 1 Elements
	             magnitudes: 1 Elements
```

**Review by calum-chamberlain — APPROVED** [review]
Looks good and sensible to me.

**Comment by calum-chamberlain** [comment]
Tests fail on `obspy/io/scardec/__init__.py` doctest - that doctest needs to be updated with a shortened magnitude.

**Comment by trichter** [comment]
Is one decimal place enough or should it be two digits for these high precision magnitudes?

**Comment by megies** [comment]
Personally, I think one decimal should be enough (and whenever I see magnitudes stated anywhere, most of the times it's one decimal) but if you think two is better, I wouldn't mind making it 2 digits.
I would assume in many cases these magnitudes just have an arbitrary amount of digits in the quakeml due to computed magnitudes being stored in there as is without being truncated.

**Comment by trichter** [comment]
I would  prefer to display two digits if two or more digits are available and to display one digit for all other cases, but this is not a strong opinion.

**Comment by megies** [comment]
I'll make it two digits, thats fine, no reason not to.

File: obspy/io/ndk/__init__.py [hunk 1]
```diff
@@ -95,11 +95,11 @@
...
 33872 Event(s) in Catalog:
-1976-01-01T01:29:53.400000Z | -29.250, -176.960 | 7.25 Mwc
-1976-01-05T02:31:44.700000Z | -13.420,  -75.140 | 5.65 Mwc
+1976-01-01T01:29:53.400000Z | -29.250, -176.960 | 7.25  Mwc
+1976-01-05T02:31:44.700000Z | -13.420,  -75.140 | 5.65  Mwc
 ...
-2010-12-31T04:11:08.400000Z | -18.980, +167.590 | 5.07 Mwc
-2010-12-31T16:30:59.700000Z |  +0.810,  -26.060 | 5.45 Mwc
+2010-12-31T04:11:08.400000Z | -18.980, +167.590 | 5.07  Mwc
+2010-12-31T16:30:59.700000Z |  +0.810,  -26.060 | 5.45  Mwc
```
Comment (trichter) [inline obspy/io/ndk/__init__.py:102]: The extra white space here and in the lines above is inconsistent.

**Comment by trichter** [comment]
Thanks. You also added an extra space for visual alignment for the single digit case. IMHO, this is not necessary, but I am probably a bit nit-pick here.

**Comment by megies** [comment]
Yeah, it's a bit ugly in case of all a decimal magnitudes, but it's less ugly than running into catalogs getting printed with ramshackle misaligned separators

**Pull request merged**